src/components/Modal/style.css

```diff
@@ -128,16 +130,20 @@
   }
 }
 
-@media (--large) {
-  .closeButton:hover {
-    background-color: #D8DADC;
+.closeButton:hover {
+  background-color: #D8DADC;
 
-    .closeButtonLabel {
-      display: block;
+  .closeButtonLabel {
+    display: block;
+
+    @media (--small-viewport) {
+      /* Safari Mobile tries to show "hover" text on click, so we make sure
+       * that mobile devices don't have any "hover text" */
+      display: none;
```

Comment:
Out of scope: We should have this for tablet size too, as it's currently breaking on iPad.

Reply:
not a regression, right?

Reply:
It was reported during an applause cycle but I was not able to reproduce it. So probably it was on an old version of Safari. Not a regression, I don't think. Probably something that has been there since we introduced the feature